pkg/apis/core/v1beta1/helper/helpers_test.go

```diff
@@ -595,365 +663,427 @@ var _ = Describe("helper", func() {
+					Expect(shouldFilter).To(Equal(expectFilterVersion))
 				},
-			}
 
-			shootMachineImage = gardencorev1beta1.ShootMachineImage{
-				Name:    "coreos",
-				Version: pointer.StringPtr("0.0.2"),
-			}
-		})
-		It("should determine that the version exists", func() {
-			exists, index := ShootMachineImageVersionExists(constraint, shootMachineImage)
-			Expect(exists).To(Equal(trueVar))
-			Expect(index).To(Equal(0))
-		})
-		It("should determine that the version does not exist", func() {
-			shootMachineImage.Name = "xy"
-			exists, _ := ShootMachineImageVersionExists(constraint, shootMachineImage)
-			Expect(exists).To(Equal(false))
-		})
-		It("should determine that the version does not exist", func() {
-			shootMachineImage.Version = pointer.StringPtr("0.0.4")
-			exists, _ := ShootMachineImageVersionExists(constraint, shootMachineImage)
-			Expect(exists).To(Equal(false))
+				// #FilterForSameMajorMinorVersion
+				Entry("Should filter version - has not the same major.minor.",
+					FilterForSameMajorMinorVersion(*semver.MustParse("1.2.0")),
+					semver.MustParse("1.1.1"),
+					gardencorev1beta1.ExpirableVersion{},
+					true,
+					false,
+				),
+				Entry("Should not filter version - has the same major.minor.",
+					FilterForSameMajorMinorVersion(*semver.MustParse("1.1.0")),
+					semver.MustParse("1.1.1"),
+					gardencorev1beta1.ExpirableVersion{},
+					false,
+					false,
+				),
+
+				// #FilterForConsecutiveMinor
+				Entry("Should filter version - has not the consecutive minor version.",
+					FilterForConsecutiveMinor(*semver.MustParse("1.3.0")),
+					semver.MustParse("1.1.1"),
+					gardencorev1beta1.ExpirableVersion{},
+					true,
+					false,
+				),
+				Entry("Should filter version - has the same minor version.",
+					FilterForConsecutiveMinor(*semver.MustParse("1.1.0")),
+					semver.MustParse("1.1.1"),
+					gardencorev1beta1.ExpirableVersion{},
+					true,
+					false,
+				),
+				Entry("Should not filter version - has consecutive minor.",
+					FilterForConsecutiveMinor(*semver.MustParse("1.1.0")),
+					semver.MustParse("1.2.0"),
+					gardencorev1beta1.ExpirableVersion{},
+					false,
+					false,
+				),
+
+				// #FilterSameVersion
+				Entry("Should filter version.",
+					FilterSameVersion(*semver.MustParse("1.1.1")),
+					semver.MustParse("1.1.1"),
+					gardencorev1beta1.ExpirableVersion{},
+					true,
+					false,
+				),
+				Entry("Should not filter version.",
+					FilterSameVersion(*semver.MustParse("1.1.1")),
+					semver.MustParse("1.1.2"),
+					gardencorev1beta1.ExpirableVersion{},
+					false,
+					false,
+				),
+
+				// #FilterExpiredVersion
+				Entry("Should filter expired version.",
+					FilterExpiredVersion(),
+					nil,
+					gardencorev1beta1.ExpirableVersion{
+						ExpirationDate: &expirationDateInThePast,
+					},
+					true,
+					false,
+				),
+				Entry("Should not filter version.",
+					FilterExpiredVersion(),
+					nil,
+					gardencorev1beta1.ExpirableVersion{},
```

Comment:
Can you add a non-expired expiration date here?
```suggestion
					gardencorev1beta1.ExpirableVersion{ExpirationDate: &metav1.Time{Time: time.Now().Add(time.Hour)}},
```